concent_api/core/tests/test_payments_report.py

```diff
@@ -0,0 +1,226 @@
+                public_key=base64.b64encode(ECCx(None).raw_pubkey)
+            )
+            client.full_clean()
+            client.save()
+            self.requestors.append(client)
+
+    def _prepare_subtasks(self):
+        for i in range(self.number_of_subtasks):
+            task_to_compute = self._get_deserialized_task_to_compute(
+                provider_public_key=self.providers[i // 3].public_key,
+                requestor_public_key=(
+                    self.requestors[(i + 1) // 3].public_key
+                    if i != (self.number_of_subtasks - 1) else self.requestors[0].public_key
+                ),
+            )
+            report_computed_task = self._get_deserialized_report_computed_task(task_to_compute=task_to_compute)
+
+            subtask = store_subtask(
+                task_id=report_computed_task.task_id,
+                subtask_id=report_computed_task.subtask_id,
+                provider_public_key=base64.b64decode(task_to_compute.provider_public_key),
+                requestor_public_key=base64.b64decode(task_to_compute.requestor_public_key),
+                state=Subtask.SubtaskState.ACCEPTED,
+                next_deadline=None,
+                task_to_compute=task_to_compute,
+                report_computed_task=report_computed_task,
+            )
+
+            self.subtasks.append(subtask)
+
+        assert Client.objects.count() == self.number_of_clients_per_type * 2
+
+    def _prepare_related_payments_mocks(self):
+        for subtask in get_subtasks_with_distinct_clients_pairs():  # pylint: disable=not-an-iterable
+            # The SCI methods are called in the same order in `core.payments.report.get_related_payments_for_pair`.
+            self.get_list_of_payments_mocks.append(
+                [
+                    self._create_batch_payment_object(
+                        self.default_payment_amount,
+                        self.closure_time,
+                    )
+                ]
+            )
+            self.get_list_of_payments_mocks.append(
+                [
+                    self._create_settlement_payment_object(
+                        self.default_payment_amount,
+                        self.closure_time,
+                    )
+                ]
+            )
+            self.get_list_of_payments_mocks.append(
+                [self._create_forced_subtask_payment_object(self.default_payment_amount, subtask.subtask_id)
+                 for subtask in Subtask.objects.filter(requestor=subtask.requestor, provider=subtask.provider)]
+            )
+            self.get_covered_additional_verification_costs_mocks.append(
+                [self._create_cover_additional_verification_costs_object(self.default_payment_amount, subtask.subtask_id)
+                 for subtask in Subtask.objects.filter(requestor=subtask.requestor, provider=subtask.provider)]
+            )
+            self.get_covered_additional_verification_costs_mocks.append(
```

Comment:
why is it called second time with exactly the same params?

Reply:
Because we want two payments to exist and be mocked, because in report they are retrieved twice in a single run - once for provider and once for requestor.